src/porting.h

```diff
@@ -220,6 +224,21 @@ inline u64 getDeltaMs(u64 old_time_ms, u64 new_time_ms)
 	return (old_time_ms - new_time_ms);
 }
 
+inline void preciseSleepUs(u64 sleep_time)
+{
+	if (sleep_time > 0)
+	{
+		u64 target_time = porting::getTimeUs() + sleep_time;
+		if (sleep_time > SLEEP_ACCURACY_US)
+			sleep_us(sleep_time - SLEEP_ACCURACY_US);
+
+		// Busy-wait the remaining time to adjust for sleep inaccuracies
+		// The target - now > 0 construct will handle overflow gracefully (even though it should
+		// never happen)
+		while ((s64)(target_time - porting::getTimeUs()) > 0);
```

Comment:
I am still suggesting this to make the infinite loop more obvious
```suggestion
		while ((s64)(target_time - porting::getTimeUs()) > 0) {}
```

Reply:
```suggestion
		s64 time_remain;
		do time_remain = target_time - porting::getTimeUs();
		while (time_remain > 0);
```

Maybe this? Or is it too verbose?

Also, this technically isn't an infinite loop since it does end.

Reply:
Too verbose IMO, just the `{}` is fine because an `;` is easily missed